p2p/discover/v4_lookup_test.go

```diff
@@ -95,6 +101,8 @@ func TestUDPv4_LookupIterator(t *testing.T) {
 	if err := checkNodesEqual(results, want); err != nil {
 		t.Fatal(err)
 	}
+	test.close()
```

Comment:
Should we remove the `defer test.close()` to avoid double close?

Reply:
Ah, I didn't see that, that's where the double-close came from (hence why I made it so double-close now is fine). Yes, good point

Reply:
I'll put them both into the existing defer instead